src/main/java/bio/terra/service/dataset/DatasetSummary.java

```diff
@@ -64,4 +69,22 @@ public DatasetSummary createdDate(Instant createdDate) {
         this.createdDate = createdDate;
         return this;
     }
+
+    public List<StorageResource> getStorage() {
+        return storage;
+    }
+
+    public DatasetSummary storage(List<StorageResource> storage) {
+        this.storage = storage;
+        return this;
+    }
+
+    public String getStorageResourceRegion(GoogleCloudResource storageResource) {
+        return this.storage.stream()
+            .filter(resource -> resource.getCloudResource().equals(storageResource.toString()))
+            .findFirst()
+            .map(StorageResource::getRegion)
+            .orElseThrow(() -> new StorageResourceNotFoundException(
+                String.format("%s could not be found for %s ", storageResource.toString(), id.toString())));
```

Comment:
unnecessary `toString()`
```suggestion
                String.format("%s could not be found for %s ", storageResource, id)));
```